### DISCUSSION TITLE
Jetson AGX Orin 64GB support?

### DISCUSSION THREAD

**Discussion by om26er** [OE4T/meta-tegra#1762] [General]
On scarthgap, is the 64GB variant supported? seems like the `jetson-agx-orin-devkit.conf` uses a different device tree `tegra234-p3737-0000+p3701-0000-nv.dtb`

What do I need to do to get that working

**Comment by madisongh** [comment]
There should be comments in the machine config files describing what hardware they support, like [this](https://github.com/OE4T/meta-tegra/blob/2b64044bc8d7affa666128c90c58cc7829f48b8c/conf/machine/p3737-0000-p3701-0005.conf#L2-L3) for the AGX Orin 64GB module.

**Reply by om26er** [reply]
@madisongh I am able to create an image for our custom yocto distribution. However the OS doesn't boot and I don't know where it fails as there is no way to see the logs. What do I need to do to see the logs from bootloader to early kernel so I know where the issue is happening. Is there supposed to be a special debug cable?

**Reply by madisongh** [reply]
If you're using the AGX Orin dev kit carrier, the serial console (debug UART) should be exposed through the micro-B USB port on the back. It's described on [this page](https://developer.nvidia.com/embedded/learn/jetson-agx-orin-devkit-user-guide/howto.html).

**Comment by om26er** [comment]
@madisongh Apologies if this should be a new thread.

How do we use initramfs? I am setting these environment variables, the image builds but it fails to load initrd
```
INITRAMFS_IMAGE_BUNDLE = "1"
INITRAMFS_IMAGE = "st-hub-initramfs"
INITRAMFS_FSTYPES = "cpio.xz"
TEGRAFLASH_INITRD_FLASH_IMAGE = ""
IMAGE_FSTYPES:pn-${INITRAMFS_IMAGE} = "${INITRAMFS_FSTYPES}"
```

This is the error `"EFI stub: ERROR: Failed to load initrd!"`

**Reply by madisongh** [reply]
That error is being reported by the EFI stub code in the kernel, which looks like it assumes the EFI bootloader always provides an initrd by default. Looking at the code, you could try adding `noinitrd` to the kernel command line to tell the stub not to do this, but I'm not sure whether the bundled initramfs would be used in that case or not.

**Reply by om26er** [reply]
I am going to drop initramfs for now and change the init binary to first mount my overlayfs (for read-only rootfs), do data partition resize and the just run systemd afterwards. Should get me started quickly.

Will visit initramfs later

**Comment by om26er** [comment]
Now I am able to boot the system correctly but it seems the date time is always set to `Thu Jan  1 00:00:00 UTC 1970` on boot.
